rebasehelper/tags.py

```diff
@@ -84,26 +116,38 @@ def sanitize(name):
                     return '{0}0'.format(tokens[0])
                 return '{0}{1}'.format(tokens[0], int(tokens[1]))
             return name.capitalize()
-        result = []
+        result: List[Tag] = []
         tag_re = re.compile(r'^(?P<prefix>(?P<name>\w+)\s*:\s*)(?P<value>.+)$')
         parsed_idx = 0
-        for section, section_content in raw_content.sections:
+        for section_index, (section, section_content) in enumerate(raw_content.sections):
             section = section.lower()
-            if not section.startswith('%package'):
+            is_list = section.startswith('%sourcelist') or section.startswith('%patchlist')
+            if not section.startswith('%package') and not is_list:
                 continue
             if parsed_idx >= len(parsed_content.sections) or section != parsed_content.sections[parsed_idx][0].lower():
                 parsed: List[str] = []
             else:
                 parsed = parsed_content.sections[parsed_idx][1]
                 parsed_idx += 1
+            if section.startswith('%sourcelist'):
+                # copy the list, we do not want to permanently modify the content
+                section_content = cls._prepend_list_with_tags(section, section_content[:], parsed, 'Source',
+                                                              cast(Tuple[Tag], tuple(result)))
+            elif section.startswith('%patchlist'):
+                section_content = cls._prepend_list_with_tags(section, section_content[:], parsed, 'Patch',
+                                                              cast(Tuple[Tag], tuple(result)))
```

Comment:
I don't like this. I think it would be better to introduce a method that would, based on `result`, return the new index, and modify the content with a simple list comprehension right here.